jgrapht-core/src/main/java/org/jgrapht/alg/treedecomposition/IntervalgraphDecomposition.java

```diff
@@ -0,0 +1,347 @@
+    private Graph<Integer, DefaultEdge> decomposition = null;
+
+    // map from decomposition nodes to the interval sets
+    private Map<Integer, Set<V>> decompositionIntervalMap = null;
+
+    // the root of the tree
+    private Integer root = null;
+
+    // input to the algorithm, list of sorted intervals
+    private List<Interval<T>> startSort, endSort;
+    private Map<Interval<T>, V> intervalToVertexMap;
+    private Map<V, Interval<T>> vertexToIntervalMap;
+
+    // helper attributes
+    private Integer currentVertex = null;
+    private Set<V> currentSet = null;
+
+    /**
+     * Private constructor for the factory methods, which changes the inputs
+     * 
+     * @param sortedByStartPoint the intervals sorted after the starting points
+     * @param sortedByEndPoint the intervals sorted after the ending points
+     * @param intervalToVertexMap maps intervals to the vertices of the graph (may be the same)
+     */
+    private IntervalgraphDecomposition(
+        List<Interval<T>> sortedByStartPoint, List<Interval<T>> sortedByEndPoint,
+        Map<Interval<T>, V> intervalToVertexMap, Map<V, Interval<T>> vertexToIntervalMap)
+    {
+        this.startSort = sortedByStartPoint;
+        this.endSort = sortedByEndPoint;
+        this.intervalToVertexMap = intervalToVertexMap;
+        this.vertexToIntervalMap = vertexToIntervalMap;
+    }
+
+    /**
+     * Factory method for creating a nice tree decomposition for interval graphs. This factory
+     * method uses general graphs and the IntervalGraphRecognizer to generate a list of intervals
+     * sorted by starting and ending points. The complexity of this method depends on the sorting
+     * algorithm in IntervalGraphRecognizer (probably quasi linear in the number of intervals)
+     * 
+     * @param graph the graph which should transformed to an interval graph and then into a
+     *        corresponding nice tree decomposition
+     * @param <V> the vertex type of the graph
+     * @param <E> the edge type of the graph
+     * @return the algorithm for the computation of the nice tree decomposition, returns null if
+     *         graph was no interval graph
+     * @see IntervalGraphRecognizer
+     */
+    public static <V, E> IntervalgraphDecomposition<Integer, V> create(Graph<V, E> graph)
+    {
+        IntervalGraphRecognizer<V, E> recog = new IntervalGraphRecognizer<>(graph);
+        HashMap<V, Interval<Integer>> vertexToIntegerMap =
+            new HashMap<>(recog.getVertexToIntervalMap().size());
+        for (Entry<V, IntervalVertexInterface<V, Integer>> entry : recog
+            .getVertexToIntervalMap().entrySet())
+            vertexToIntegerMap.put(entry.getKey(), entry.getValue().getInterval());
+        if (recog.isIntervalGraph())
+            return new IntervalgraphDecomposition<Integer, V>(
+                recog.getIntervalsSortedByStartingPoint(),
+                recog.getIntervalsSortedByStartingPoint(), recog.getIntervalToVertexMap(),
```

Comment:
shouldn't this be recog.getIntervalsSortedByEndingPoint() as in the constructor indicated?

Reply:
I guess my testing was too bad.

Reply:
more tests from my side are still on their way. :)